naptime/src/main/scala/org/coursera/naptime/ari/engine/EngineImpl.scala

```diff
@@ -282,8 +282,10 @@ class EngineImpl @Inject() (
 
               // MultiGets return a list of ids, and Gets return a single id (or null)
               val intersection = responseIds.filter(id => ids.contains(id.toString))
+              val indexedIds = ids.zipWithIndex.toMap
+              val sortedIntersection = intersection.sortBy(id => indexedIds(id.toString))
```

Comment:
same comment here -- is it ever possible that id wouldn't be found in indexedIds? might be safer to do `.get` and then a `.getOrElse(-1)` or something